pkg/skaffold/render/renderer/renderer.go

```diff
@@ -45,26 +45,21 @@ const (
 )
 
 type Renderer interface {
-	Render(context.Context, io.Writer, []graph.Artifact) error
+	Render(context.Context, io.Writer, []graph.Artifact, bool, string) error
 }
 
 // NewSkaffoldRenderer creates a new Renderer object from the latestV2 API schema.
 func NewSkaffoldRenderer(config *latestV2.RenderConfig, workingDir string) (Renderer, error) {
 	// TODO(yuwenma): return instance of kpt-managed mode or skaffold-managed mode defer to the config.Path fields.
 	// The alpha implementation only has skaffold-managed mode.
-	// TODO(yuwenma): The current work directory may not be accurate if users use --filepath flag.
-	hydrationDir := filepath.Join(workingDir, DefaultHydrationDir)
-
-	generator := generate.NewGenerator(workingDir, config.Generate)
-	/* TODO(yuwenma): Apply new UX
-		if config.Generate == nil {
-		// If render.generate is not given, default to current working directory.
-		defaultManifests := filepath.Join(workingDir, "*.yaml")
-		generator = generate.NewGenerator(workingDir, latestV2.Generate{Manifests: []string{defaultManifests}})
+	var hydrationDir string
+	if config.Output == "" {
+		hydrationDir = filepath.Join(workingDir, DefaultHydrationDir)
```

Comment:
does the hydration directory necessarily have to be inside the working directory? I think that the current precedence in skaffold is to put intermediate output inside `$HOME/.skaffold/foo` directory. 
If the users are expected to interact with the rendered manifests why is the default a hidden directory, or is this not the final destination?

Reply:
Really good catch. This is not the final destination, the config.Output will be deprecated and change to use the --output flag instead (so as some other UX names). I'm working on the PR now, hope you can see something make more sense. 😄